packages/kit/src/runtime/client/client.js

```diff
@@ -1038,21 +1038,37 @@ export function create_client(app, target) {
 	}
 
 	/**
-	 * @param {URL} url
+	 * @param {URL | undefined} original_url
 	 * @param {boolean} invalidating
 	 */
-	function get_navigation_intent(url, invalidating) {
-		if (is_external_url(url, base)) return;
+	function get_navigation_intent(original_url, invalidating) {
+		if (!original_url) return undefined;
+
+		// rewriteUrl could alter the given URL, so we pass a copy
+		let rewritten_url;
+		try {
+			rewritten_url = app.hooks.rewriteUrl({ url: new URL(original_url) });
+		} catch (e) {
```

Comment:
I'm not sure we need to catch this? @Rich-Harris just did an error handling overhaul, so might have an opinion on this. Rich, if you look at this you should also check the server-side file - maybe we do need to keep the `catch` there?

Reply:
I just tested it, and if we don't catch there any error thrown in `rewriteUrl` while running in the browser crashes the client